packages/core/test/lib/commands/compile.js

```diff
@@ -8,11 +8,47 @@ const command = require("../../../lib/commands/compile");
 const path = require("path");
 const fs = require("fs-extra");
 const glob = require("glob");
-let config;
-let output = "";
-let memStream;
+
+const makeResolveSource = config => async contractSourcePath => {
+  const { filePath } = await config.resolver.resolve(
+    contractSourcePath,
+
+    // to pretend we're inside the contracts directory
+    path.join(config.contracts_directory, "_")
+  );
+
+  return filePath;
+};
+
+const makeUpdateSources = config => {
+  const resolveSource = makeResolveSource(config);
+
+  return async contractSourcePaths => {
+    // convert (e.g.) "ConvertLib.sol" to "/path/to/contracts/ConvertLib.sol"
+    const filePaths = await Promise.all(contractSourcePaths.map(resolveSource));
+
+    const stats = {};
+    for (const filePath of filePaths) {
+      stats[filePath] = fs.statSync(filePath);
+
+      var newTime = new Date().getTime();
+      fs.utimesSync(filePath, newTime, newTime);
+    }
+
+    return () => {
+      for (const [filePath, stat] of Object.entries(stats)) {
+        fs.utimesSync(filePath, stat.atime, stat.mtime);
+      }
+    };
+  };
+};
```

Comment:
If this is worth keeping, we'll probably want to put it into a separate file? It's kinda ugly.